MedtrumKit/PumpManager/PeripheralManager.swift

```diff
@@ -88,13 +88,20 @@ class PeripheralManager : NSObject {
 extension PeripheralManager {
     
     // Connect step 1
-    private func doAuthorize() async {
+    private func doAuthorize(withPreviousToken: Bool = false) async {
+        let sessionToken = !withPreviousToken ? self.pumpManager.state.sessionToken : self.pumpManager.state.previousSessionToken
```

Comment:
Hmm why is this needed? This should be prevented during the activation process imo. e.g. by preventing creation of a new token when you are connected already to the patch. and/or ensuring you don't connect to the patch when you are in a deactivated state